assets/js/components/ClusterDetails/ClusterDetailsNew.jsx

```diff
@@ -0,0 +1,280 @@
+import React, { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+
+import Button from '@components/Button';
+
+import ListView from '@components/ListView';
+import Pill from '@components/Pill';
+import Table from '@components/Table';
+import Tooltip from '@components/Tooltip';
+import TriggerChecksExecutionRequest from '@components/TriggerChecksExecutionRequest';
+
+import { groupBy } from '@lib/lists';
+
+import SiteDetails from './SiteDetails';
+
+import { getClusterName } from '@components/ClusterLink';
+import HostLink from '@components/HostLink';
+
+import { EOS_SETTINGS, EOS_CLEAR_ALL, EOS_PLAY_CIRCLE } from 'eos-icons-react';
+import { getCluster } from '@state/selectors';
+import { getLastExecution } from '@state/selectors/lastExecutions';
+import classNames from 'classnames';
+import ChecksResultOverviewNew from '@components/ClusterDetails/ChecksResultOverviewNew';
+import ProviderLabel from './ProviderLabel';
+
+export const truncatedClusterNameClasses = classNames(
+  'font-bold truncate w-60 inline-block align-top'
+);
+
+const siteDetailsConfig = {
+  usePadding: false,
+  columns: [
+    {
+      title: 'Hostname',
+      key: '',
+      render: (_, hostData) => (
+        <HostLink hostId={hostData.hostId}>{hostData.name}</HostLink>
+      ),
+    },
+    { title: 'Role', key: 'hana_status' },
+    {
+      title: 'IP',
+      key: 'ips',
+      className: 'table-col-m',
+      render: (content) => content?.join(', '),
+    },
+    {
+      title: 'Virtual IP',
+      key: 'virtual_ip',
+      className: 'table-col-m',
+    },
+    {
+      title: '',
+      key: '',
+      className: 'table-col-xs',
+      render: (_, item) => {
+        const { attributes, resources } = item;
+        return <SiteDetails attributes={attributes} resources={resources} />;
+      },
+    },
+  ],
+};
+
+const getStatusPill = (status) =>
+  status === 'healthy' ? (
+    <Pill className="bg-green-200 text-green-800 mr-2">Healthy</Pill>
+  ) : (
+    <Pill className="bg-red-200 text-red-800 mr-2">Unhealthy</Pill>
+  );
+
+export const ClusterDetailsNew = () => {
+  const { clusterID } = useParams();
+  const navigate = useNavigate();
+
+  const cluster = useSelector(getCluster(clusterID));
+
+  const dispatch = useDispatch();
+
+  const lastExecution = useSelector(getLastExecution(clusterID));
+
+  useEffect(() => {
+    dispatchUpdateLastExecution();
+  }, [dispatch]);
+
+  const dispatchUpdateLastExecution = () => {
+    dispatch({
```

Comment:
Let's put this constant action in a actions files, in the state. So we don't need to have this `dispatchUpdateLastExecution` function and we can simply do `dispatch(updateLastExecution(groupID))` or similar.
What do you think?
we started doing this in one of the previous changes

Reply:
done